File: typescript/packages/management/src/tools/update_agent_model_settings.ts
import type { ToolContext, ToolHandler } from '../registrar.js';

export const updateAgentModelSettings: ToolHandler = async (
  ctx: ToolContext,
  input: Record<string, unknown>,
) => {
  const slug = String(input.slug);
  const patch: Record<string, unknown> = {};
  if (input.backend === 'anthropic' || input.backend === 'google') {
    patch.backend = input.backend;
  }
  if (typeof input.model === 'string') {
    patch.model = input.model;
  }
  if (typeof input.temperature === 'number') {
    patch.temperature = input.temperature;
  }
  if (typeof input.max_tokens === 'number') {
    patch.max_tokens = input.max_tokens;
  }
  if (typeof input.max_turns === 'number') {
    patch.max_turns = input.max_turns;
  }
  if (typeof input.timeout_s === 'number') {
    patch.timeout_s = input.timeout_s;
  }
  const agent = await ctx.client.agents.updateModelSettings(
    ctx.sandboxId,
    slug,
    patch as Parameters<typeof ctx.client.agents.updateModelSettings>[2],
  );
  return { agent };
};

File: python/copass-management/src/copass_management/tools/update_agent_model_settings.py
from __future__ import annotations

from typing import Any, Dict, TYPE_CHECKING

if TYPE_CHECKING:
    from copass_management.registrar import ToolContext


async def update_agent_model_settings(
    ctx: "ToolContext", input: Dict[str, Any],
) -> Any:
    slug = str(input["slug"])
    kwargs: Dict[str, Any] = {}
    if "backend" in input and input["backend"] is not None:
        kwargs["backend"] = str(input["backend"])
    if "model" in input and input["model"] is not None:
        kwargs["model"] = str(input["model"])
    if "temperature" in input and input["temperature"] is not None:
        kwargs["temperature"] = float(input["temperature"])
    if "max_tokens" in input and input["max_tokens"] is not None:
        kwargs["max_tokens"] = int(input["max_tokens"])
    if "max_turns" in input and input["max_turns"] is not None:
        kwargs["max_turns"] = int(input["max_turns"])
    if "timeout_s" in input and input["timeout_s"] is not None:
        kwargs["timeout_s"] = int(input["timeout_s"])
    agent = await ctx.client.agents.update_model_settings(
        ctx.sandbox_id, slug, **kwargs,
    )
    return {"agent": agent}
